Cargo.toml

```diff
@@ -48,3 +51,4 @@ debug = true
 
 [profile.dev]
 panic = "abort"
+lto = true
```

Comment:
Is there a reason for switching link time optimizations on?

Reply:
Yes, we had a problem where we were only able to build in release and not debug (ld errors) and we found out that turning on lto on debug solved the problem. I don't remember what the exact problem was but there is maybe another solution

Reply:
Sorry for reopening that, but I just tried and I am able to build in debug mode without lto.

The additional link time introduced is not insignificant (with 64 cores/jobs +40s out of a 145s build). I'd rather not keep that flag in the dev profile if it's not necessary.

Reply:
You're right, I tried and now it works (boringssl202311 and boringssl202403 with and without asan)